enhancements/machine-config/mco-cgroupsv2-support.md

```diff
@@ -0,0 +1,220 @@
+```yaml
+  kernelArguments:
+    - systemd.unified_cgroup_hierarchy=1
+    - cgroup_no_v1="all"
+    - psi=1 
+```
+
+#### Failure Modes
+
+N/A
+
+#### Support Procedures
+
+### User Stories
+
+#### 1. As a user, I would like to install a cluster that uses cgroup v2
+
+### Risks and Mitigations
+
+The primary risk of cgroup v2 is some workloads not supporting the changed
+cgroup filesystem paths. This is the reason why it is off by default and has to
+be enabled at install time.
+
+## Design Details
+
+### Test Plan
+
+Testing should be thoroughly done at all levels, including unit, end-to-end, and
+integration.
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+cgroup v2 will be dev preview on its initial release. Internal and customer
+usage will be critical to gather information on bugs and enhancements to the
+underlying subsystem.
+
+Graduation requirements to Tech Preview are:
+
+* No regressions from cgroup v1 to cgroup v2
+* Processes are correctly OOMed
+* CPU management works as expected
+* No performance issues - PSAP and QE teams will be asked to test their suites for regressions
+* Metrics are accurate and correctly submitted to monitoring
+
+#### Tech Preview -> GA
+
+With sufficient internal testing and customer feedback the feature will graduate
+to Tech Preview.
+
+Upon graduation to GA the feature will still be turned off by default until we define another enhancement to specify how cgroup v2 is enabled by default within OpenShift.
+
+Graduation requirements to GA:
+* Upstream graduation to GA
+* Internal stakeholders are using cgroup v2 without issue
+* Tech Preview Graduation requirements are still good
+* Add blocking cgroup v2 upgrade jobs
+* CI OpenShift cgroup v2 upgrade jobs pass percentage is similar or better than the OpenShift cgroup v1 upgrade job pass percentage
+
+The following jobs will be run against cgroup v2 periodically and with a minimum of 100 runs:
+
+- periodic-ci-openshift-release-master-nightly-4.10-e2e-aws-upgrade
+- periodic-ci-openshift-release-master-ci-4.10-e2e-azure-ovn-upgrade
+- periodic-ci-openshift-release-master-ci-4.10-upgrade-from-stable-4.9-e2e-gcp-ovn-upgrade
+- periodic-ci-openshift-release-master-ci-4.10-e2e-aws-ovn-upgrade
+- periodic-ci-openshift-release-master-ci-4.10-upgrade-from-stable-4.9-e2e-aws-ovn-upgrade
+- periodic-ci-openshift-release-master-ci-4.10-upgrade-from-stable-4.9-e2e-azure-upgrade
+- periodic-ci-openshift-release-master-ci-4.10-e2e-gcp-upgrade
+- periodic-ci-openshift-release-master-nightly-4.10-upgrade-from-stable-4.9-e2e-metal-ipi-upgrade-ovn-ipv6
+- periodic-ci-openshift-release-master-ci-4.10-upgrade-from-stable-4.9-e2e-vsphere-upgrade
+
+### Upgrade / Downgrade Strategy
+
+Downgrading a cluster to an OpenShift version not containing cgroup v2 support
+is unsupported.
+
+### Version Skew Strategy
+
+A cluster installed with cgroup v2 would not be impacted by the version skew
```

Comment:
you will be when "Add blocking cgroup v2 upgrade jobs" is done for your GA criteria.  You plan to revisit?

Reply:
fixed.